lib/plug/csrf_protection.ex

```diff
@@ -103,17 +103,34 @@ defmodule Plug.CSRFProtection do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verify_opts opts
     csrf_token = get_session(conn, "_csrf_token")
     Process.put(:plug_csrf_token, csrf_token)
-
     if not verified_request?(conn, csrf_token) do
-      raise InvalidCSRFTokenError
+      if opts[:with] == :exception do
+        raise InvalidCSRFTokenError
+      else 
+        conn = clear_session conn
+      end
     end
 
     register_before_send(conn, &ensure_same_origin_and_csrf_token!(&1, csrf_token))
   end
 
+  
+  ## verifies if options passed are valid
+  ## :with - should be one of :exception or :nil_session, defaults to :exception     
+  defp verify_opts opts do
```

Comment:
Was reading this PR again and just realized it would be better to remove this `verify_opts/1` function since we don't require any opts to be passed.